wakuv2/waku.go

```diff
@@ -1800,15 +1777,40 @@ func (w *Waku) StopDiscV5() error {
 }
 
 func (w *Waku) ConnectionChanged(state connection.State) {
-	if !state.Offline && !w.onlineChecker.IsOnline() {
-		select {
-		case w.goingOnline <- struct{}{}:
-		default:
-			w.logger.Warn("could not write on connection changed channel")
+	isOnline := !state.Offline
+	if w.cfg.LightClient {
+		//TODO: Update this as per  https://github.com/waku-org/go-waku/issues/1114
+		//Trigger FilterManager to take care of any pending filter subscriptions
+		go w.filterManager.onConnectionStatusChange(w.cfg.DefaultShardPubsubTopic, isOnline)
+
+		//If connection state is reported by something other than peerCount becoming 0 e.g from mobile app, disconnect all peers
+		if (state.Offline && len(w.node.Host().Network().Peers()) > 0) ||
+			(w.state.Type != state.Type && !w.state.Offline && !state.Offline) { // network switched between wifi and cellular
+			w.logger.Info("connection switched or offline detected via mobile, disconnecting all peers")
+			w.node.DisconnectAllPeers()
+			if w.cfg.LightClient {
+				w.filterManager.networkChange()
+			}
 		}
 	}
 
-	w.onlineChecker.SetOnline(!state.Offline)
+	if isOnline && !w.onlineChecker.IsOnline() {
+		if !w.cfg.LightClient {
+			//TODO: analyze if we need to discover and connect to peers with peerExchange loop enabled.
+			w.discoverAndConnectPeers()
+			select {
+			case w.goingOnline <- struct{}{}:
+			default:
+				w.logger.Warn("could not write on connection changed channel")
+			}
+
+		}
+	}
+	if !w.cfg.LightClient {
```

Comment:
`if !w.cfg.LightClient` and `if w.cfg.LightClient` existed in 3 places in this method, wondering if it's possible to improve it with only one `if w.cfg.LightClient else ...` ?

Reply:
yea, this method looks little messy..i don't want to loose ordering of the statements by trying to group them as it will cause some other issues. Let me try to see what i can do.